service/history/mutableStateBuilder.go

```diff
@@ -678,6 +677,14 @@ func (e *mutableStateBuilder) FailDecision() {
 	e.UpdateDecision(failDecisionInfo)
 }
 
+func (e *mutableStateBuilder) clearStickyness() {
+	e.executionInfo.StickyTaskList = ""
+	e.executionInfo.StickyScheduleToStartTimeout = 0
+	e.executionInfo.ClientLibraryVersion = ""
```

Comment:
does not feel right to tie the client version to stickiness

Reply:
why?

from the lease of sticky tasklist, all new worker will use sticky by default.
which means if server decide to put anything in the sticky tasklist, server need to be sure that  the worker can handle it.

the ordinary tasklist is used when a worker first poll from server.
and at that time, server can only assume that worker has the very basic functionality.